src/test/java/com/training/web/tests/FlipkartTest.java

```diff
@@ -172,46 +175,52 @@ public void testAddToCartFunctionality(String product) {
         softAssert.assertThat(Integer.parseInt(cartPage.getShoePriceTotalInCart()) == totalPriceOfShoeSelectedFromList).isTrue();
     }
 
-
     /**
      * @param :accepting shoes from json
-     * Below testcase is clicking policies and checking the header and back to top button
+     *                   Below testcase is clicking policies and checking the header and back to top button
      */
     @Test
     public void testPolicyBackToTop() {
-        ArrayList<String> policyElementsWhichNeedToClick = new ArrayList<>(Arrays.asList(map.get("policyElementsWhichNeedToClick").split(",")));
-        ArrayList<String> policyHeaders = new ArrayList<>(Arrays.asList(map.get("policyHeaders").split(",")));
-        int counter = 0;
+        ArrayList<String> policyElementsWhichNeedToClick =
+                new ArrayList<>(Arrays.asList(map.get("policyElementsWhichNeedToClick").split(",")));
+        ArrayList<String> policyHeaders =
+                new ArrayList<>(Arrays.asList(map.get("policyHeaders").split(",")));
 
+        AtomicInteger counter = new AtomicInteger();
         HomePage homePage = new HomePage().popUpCancel();
 
-        for (String policyHeader : policyHeaders) {
-            PolicySubPage policySubPage = homePage.clickPolicySingleElement(policyElementsWhichNeedToClick.get(counter));
+        policyHeaders.stream().forEach((policyHeader) -> {
```

Comment:
Whenever you want to play with indexes - always use `IntStream.range()` or `rangeClosed() `
Then you would need to define a counter.